pkg/sdk/vault/client.go

```diff
@@ -303,7 +311,7 @@ func NewClientFromRawClient(rawClient *vaultapi.Client, opts ...ClientOption) (*
 					// Projected SA tokens do expire, so we need to move the reading logic into the loop
 					jwt, err := ioutil.ReadFile(serviceAccountFile)
 					if err != nil {
-						logger.Errorf("failed to read SA token %s: %v", serviceAccountFile, err.Error())
+						client.logger.Error("failed to read SA token", map[string]interface{}{"sa": serviceAccountFile, "err": err})
```

Comment:
```suggestion
						client.logger.Error("failed to read SA token", map[string]interface{}{"serviceAccount": serviceAccountFile, "err": err})
```

Descriptive names are better IMHO.